rails_app/app/controllers/currencies_controller.rb

```diff
@@ -41,6 +41,16 @@ def all
     end
   end
 
+  def edit_favorite
+    @user = current_user
+  end
+
+  def change_favorite
+    if current_user.update favorite_currency_id: params[:user][:favorite_currency_id]
```

Comment:
Since in #3 now you've set `favorite_currency` has an association in the `User` model, this could be done in a different way:

```ruby
currency = Currency.find(id: params[:user][:favorite_currency_id])
current_user.update(favorite_currency: currency)

Reply:
This way, if a unrecognised currency id is submitted, it will fail.

Reply:
:+1:

Reply:
Adressed in d4fdea0ca5a96a4648e1e08fe6f7f760212c8d2f